recording/lib/reporter.ex

```diff
@@ -120,4 +142,34 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording.Reporter do
 
     {:reply, report, state}
   end
+
+  defp add_wallclock_start_time(track, rtcp) do
+    delta_t_ns =
+      (rtcp.sender_info.rtp_timestamp - track.start_timestamp) / track.clock_rate * 10 ** 9
+
+    start_timestamp_wallclock = rtcp.sender_info.wallclock_timestamp - delta_t_ns
+
+    Map.put(track, :start_timestamp_wallclock, start_timestamp_wallclock)
+  end
+
+  defp recalculate_offsets(tracks) do
+    {tracks, _acc} =
+      tracks
+      |> Enum.sort_by(fn {_filename, track} -> track.offset end)
+      |> Enum.map_reduce(nil, fn {filename, track}, acc ->
+        cond do
+          not Map.has_key?(track, :start_timestamp_wallclock) ->
+            {{filename, track}, acc}
+
+          is_nil(acc) ->
+            {{filename, track}, track}
+
+          true ->
+            offset = acc.offset + track.start_timestamp_wallclock - acc.start_timestamp_wallclock
+            {{filename, %{track | offset: offset}}, acc}
+        end
+      end)
+
+    tracks
+  end
```

Comment:
Some comments are needed about how this synchronization works and why we do it that way.